lib/pluggy_elixir/transaction.ex

```diff
@@ -0,0 +1,214 @@
+    :account_id,
+    :provider_code,
+    :status,
+    :payment_data
+  ]
+
+  @type t :: %__MODULE__{
+          id: binary(),
+          description: binary(),
+          description_raw: binary(),
+          currency_code: binary(),
+          amount: float(),
+          date: NaiveDateTime.t(),
+          balance: float(),
+          category: binary(),
+          account_id: binary(),
+          provider_code: binary(),
+          status: binary(),
+          payment_data: payment_data()
+        }
+
+  @type payment_data :: %{
+          payer: identity(),
+          receiver: identity(),
+          reason: binary(),
+          payment_method: binary(),
+          reference_number: binary()
+        }
+
+  @type identity :: %{
+          type: binary(),
+          branch_number: binary(),
+          account_number: binary(),
+          routing_number: binary(),
+          document_number: document()
+        }
+
+  @type document :: %{
+          type: binary(),
+          value: binary()
+        }
+
+  @transactions_path "/transactions"
+  @default_page_size 20
+  @default_page_number 1
+
+  @doc """
+  List transactions supporting filters (by account and period) and pagination.
+
+  ### Examples
+
+      iex> params = %{
+        account_id: "03cc0eff-4ec5-495c-adb3-1ef9611624fc",
+        from: ~D[2021-04-01],
+        to: ~D[2021-05-01],
+        page_size: 100,
+        page: 1
+      }
+      iex> Transaction.all_by_account(params)
+      {:ok,
+       %{
+         page: 1,
+         total: 1,
+         total_pages: 1,
+         transactions: [
+           %Transaction{
+             account_id: "03cc0eff-4ec5-495c-adb3-1ef9611624fc",
+             amount: 1_500,
+             balance: 3_000,
+             category: "Transfer",
+             currency_code: "BRL",
+             date: ~N[2021-04-12 00:00:00.000],
+             description: "TED Example",
+             description_raw: nil,
+             id: "6ec156fe-e8ac-4d9a-a4b3-7770529ab01c",
+             payment_data: %{
+               payer: %{
+                 account_number: "1234-5",
+                 branch_number: "090",
+                 document_number: %{
+                   type: "CPF",
+                   value: "882.937.076-23"
+                 },
+                 routing_number: "001",
+                 type: nil
+               },
+               payment_method: "TED",
+               reason: "Taxa de serviço",
+               receiver: %{
+                 account_number: "9876-1",
+                 branch_number: "999",
+                 document_number: %{
+                   type: "CNPJ",
+                   value: "08.050.608/0001-32"
+                 },
+                 routing_number: "002",
+                 type: nil
+               },
+               reference_number: "123456789"
+             },
+             provider_code: "123",
+             status: "POSTED"
+           }
+         ]
+       }}
+  """
+
+  @spec all_by_account(
+          %{
```

Comment:
What do you think about creating a type for these params and the response?

```elixir
  @typep params :: %{
           :account_id => String.t(),
           :from => Date.t(),
           :to => Date.t(),
           optional(:page_size) => integer(),
           optional(:page) => integer()
         }

  @typep ok_response :: %{page: integer(), total_pages: integer(), total: integer(), transactions: [t()]}
  @typep error_response :: PluggyElixir.HttpClient.Error.t() | String.t()
  @typep response :: {:ok, ok_response()} | {:error, error_response()}
```

```suggestion
          params()
```